[TEMT] Added Temt6000 light sensor using a d1 mini

diepeterpan opened the pull request:
## Description:
Added Temt6000 light sensor module using a d1 mini

![image](https://user-images.githubusercontent.com/15066052/226084328-5be56d61-d968-47f3-b177-ce441579fe75.png)
![image](https://user-images.githubusercontent.com/15066052/226084332-8f3c4778-d649-44f4-8a9a-c6f7922d0a36.png)
![image](https://user-images.githubusercontent.com/15066052/226084342-39c60a0e-7704-4aac-b3de-18b527900557.png)

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

diepeterpan committed:
Added light sensor TEMPT6000

Added light sensor TEMPT6000

1technophile commented on environments.ini:
```diff
@@ -1675,3 +1676,62 @@ build_flags =
+[env:esp01-hcsr501-ms]
+platform = ${com.esp8266_platform}
+board = esp01_1m
+board_build.ldscript = eagle.flash.1m64.ld
+lib_deps =
+  ${com-esp.lib_deps}
+  ${libraries.wifimanager8266}
+  ${libraries.unifiedsensor}
+  ${libraries.esp8266_mdns}
+build_flags =
+  ${com-esp.build_flags}
+  '-UZmqttDiscovery'          ; disables MQTT Discovery
+;  '-UZmqttDiscovery="HADiscovery"'
+  '-DvalueAsATopic=true'       ; MQTT topic includes model and device
+  '-DHCSR501_LED_NOTIFY_GPIO=2'
+  '-DHCSR501_LED_ON=LOW'
+  '-UMQTTsetMQTT' ;We remove this function to have sufficient FLASH available for OTA, you should also use ESPWifiManualSetup and 'board_build.ldscript = eagle.flash.1m64.ld' to save flash memory and have OTA working
+  '-UMQTT_HTTPS_FW_UPDATE' ;We remove this function to have sufficient FLASH available for OTA, you should also use ESPWifiManualSetup and 'board_build.ldscript = eagle.flash.1m64.ld' to save flash memory and have OTA working
+;  '-DGateway_Name="OpenMQTTGateway_CSR501_MS"'
+  '-DZsensorHCSR501="HCSR501"'
+  '-DHCSR501_GPIO=0'
+  '-DGateway_Name="OpenMQTTGatewayMS"'
+  '-DESPWifiManualSetup=true'
+  '-Dwifi_ssid=""'
+  '-Dwifi_password=""'
+  '-DMQTT_USER=""'
+  '-DMQTT_PASS=""'
+  '-DMQTT_SERVER="..."'
+board_build.flash_mode = dout
+custom_description = Motion sensor gateway 
+custom_hardware = Motion sensor first version
+
+[env:esp01-hcsr501-ms-ota-test]
+extends = env:esp01-hcsr501-ms
+upload_protocol = espota
+upload_port = 127.0.0.1
+upload_flags =
+  --auth=OTAPASSWORD
+  --port=8266
+  
```
Could you add ` '-DZsensorTEMT6000="TEMT6000"'` to `esp32dev-all-test` so that the continuous integration will check that it builds with ESP32?

We should not add three other environments. The users should store this kind of custom environment in a prod_env.ini
By adding the module into the `nodemcu-all-test` and the `esp32dev-all-test` we verify that the module builds correctly and can be used by changing the configuration of the gateway through Platformio.
I would like to avoid being in a situation on which we store all possible configuration in the web upload and make it difficult for other users to find the right binary.

diepeterpan replied:
All good, done.

diepeterpan committed:
add '-DZsensorTEMT6000="TEMT6000"' to esp32dev-all-test

We should not add three other environments. The users should store this kind of custom environment in a prod_env.ini
By adding the module into the nodemcu-all-test and the esp32dev-all-test we verify that the module builds correctly and can be used by changing the configuration of the gateway through Platformio.

1technophile commented on platformio.ini:
```diff
@@ -101,6 +101,8 @@ extra_configs =
 ;default_envs = lolin_c3_mini
 ;default_envs = thingpulse-espgateway
 ;default_envs = thingpulse-espgateway-ethernet
+;default_envs = d1mini-temt6000-ls
```
Just remove these 2 also and we will be good to go

diepeterpan replied:
No problem, done.

diepeterpan committed:
Default envs to remove

1technophile commented:
Thanks
